pkg/controller/vsphere/reconciler.go

```diff
@@ -117,9 +129,15 @@ func (r *Reconciler) update() error {
 
 	// TODO: we won't always want to reconcile power state
 	//  but as per comment in clone() function, powering on right on creation might be problematic
-	if ok, err := vm.reconcilePowerState(); err != nil || !ok {
+	ok, task, err := vm.reconcilePowerState()
+	if err != nil || !ok {
 		return err
 	}
+
+	if err = setProviderStatus(task, conditionSuccess(), r.machineScope, vm); err != nil {
+		return errors.Wrap(err, "Failed to set provider status")
+	}
```

Comment:
I wonder if this should be called from within `reconcileMachineWithCloudState` any thoughts?

Reply:
makes sense, fixed